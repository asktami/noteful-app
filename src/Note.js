import React from 'react';
import NoteItem from './NoteItem';

import config from './config';
import NotefulContext from './NotefulContext';

//
// QUESTION - why get
// TypeError: Cannot read property 'context' of undefined
// when click on note title???
// error showing as on both FolderItem and Note
//

const Note = props => {
	const { notes } = this.context;

	// console.log('Note props = ', JSON.stringify(props));
	// console.log('Note params = ', props.match.params);
	// console.log('Note note id = ', props.match.params.noteId);

	const note = notes.find(note => note.id == props.match.params.noteId);

	return (
		<article>
			<div className="note">
				<NoteItem note={note} togglePopup={props.togglePopup} />
				<p className="note-content">{note.content}</p>
			</div>
		</article>
	);
};

export default Note;
