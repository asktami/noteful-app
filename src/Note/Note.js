import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import NoteItem from '../NoteItem/NoteItem';

import NotefulContext from '../NotefulContext';

const Note = props => {
	// need to grab NotefulContext (globals)
	const context = useContext(NotefulContext);
	const { notes } = context;

	// NOTE
	// reloading the browser wipes the context from memory
	// so need to handle if user reloads the browser and its undefined
	// done by adding || {} here AND : '' for props.modified in NoteItem too
	const note = notes.find(note => note.id === props.match.params.noteId) || {};

	return (
		<section>
			<div className="note">
				<NoteItem {...props} note={note} />
				<p className="note-content">{note.content}</p>
			</div>
		</section>
	);
};

export default Note;

// to catch bugs
// check that get a notes array that has id, name, content, and modified
// this array is the "notes" variable coming from context
Note.propTypes = {
	notes: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string.isRequired,
			folderId: PropTypes.string.isRequired,
			name: PropTypes.string.isRequired,
			content: PropTypes.string.isRequired,
			modified: PropTypes.instanceOf(Date).isRequired
		})
	)
};
