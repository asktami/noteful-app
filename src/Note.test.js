import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Note from './Note';

describe(`Note component`, () => {
	const props = {
		match: {
			params: {
				noteId: 1
			}
		},
		notes: [
			{
				id: 1,
				folderId: 101,
				title: 'test-note-title',
				modified: new Date(2019, 8, 30)
			}
		]
	};

	it('renders a Note by default', () => {
		const wrapper = shallow(<Note {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders the Note given props', () => {
		const wrapper = shallow(<Note {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
