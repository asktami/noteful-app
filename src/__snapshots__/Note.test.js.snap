// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders a Note by default 1`] = `
<div
  className="note"
>
  <NoteItem
    note={
      Object {
        "folderId": 101,
        "id": 1,
        "modified": 2019-09-30T04:00:00.000Z,
        "title": "test-note-title",
      }
    }
  />
  <p
    className="note-content"
  />
</div>
`;

exports[`Note component renders the Note given props 1`] = `
<div
  className="note"
>
  <NoteItem
    note={
      Object {
        "folderId": 101,
        "id": 1,
        "modified": 2019-09-30T04:00:00.000Z,
        "title": "test-note-title",
      }
    }
  />
  <p
    className="note-content"
  />
</div>
`;
